hm_pyhelper/hardware_definitions.py

```diff
@@ -1,30 +1,79 @@
 import os
+from enum import Enum, auto
+from collections import namedtuple
 from hm_pyhelper.exceptions import UnknownVariantException, \
-                                   UnknownVariantAttributeException
+    UnknownVariantAttributeException
 
+BALENA_ENV_RASPBERRY_PI_MODELS = [
+    'raspberry-pi2',
+    'raspberrypi3',
+    'raspberrypi3-64',
+    'raspberrypi4-64',
+    'nebra-hnt',
+    'raspberrypicm4-ioboard'
+]
 
-def is_rockpi():
-    return 'rockpi-4b-rk3399' in os.getenv('BALENA_DEVICE_TYPE')
+BALENA_ENV_ROCKPI_MODELS = ['rockpi-4b-rk3399']
 
+SBCInfo = namedtuple('SBCInfo', ['vendor_id', 'vendor_name', 'model_name'])
 
-def is_raspberry_pi():
+
+class DeviceVendor(Enum):
+    """
+    Enum for device vendors.
+    """
+    INVALID = auto()
+    ROCK_PI = auto()
+    RASPBERRY_PI = auto()
+
+
+def device_model():
+    with open('/proc/device-tree/model', 'r') as f:
+        return f.readline().strip()
+
+
+def sbc_info() -> SBCInfo:
+    '''
+    return SBCInfo formed by reading '/proc/device-tree/model'
+    '''
+    sbc_info = SBCInfo(vendor_id=DeviceVendor.INVALID, vendor_name='', model_name='')
+    dev_model = device_model()
+    if dev_model.lower().find('raspberry') >= 0:
+        sbc_info = SBCInfo(vendor_id=DeviceVendor.RASPBERRY_PI,
+                           vendor_name='Raspberry Pi',
+                           model_name=dev_model)
+    elif dev_model.lower().find('rock') >= 0:
+        sbc_info = SBCInfo(vendor_id=DeviceVendor.ROCK_PI,
+                           vendor_name='Radxa Rock Pi',
+                           model_name=dev_model)
+    return sbc_info
+
+
+def is_rockpi() -> bool:
+    device_type = os.getenv('BALENA_DEVICE_TYPE')
+
+    # use device tree supplied model name if evn not set
+    if not device_type:
+        return sbc_info().vendor_id == DeviceVendor.ROCK_PI
+
+    # honor env override
+    if device_type in BALENA_ENV_ROCKPI_MODELS:
```

Comment:
These three lines could be simplified to `return device_type in BALENA_ENV_ROCKPI_MODELS`

Reply:
done.

Reply:
Also unchanged I think you forgot to push

Reply:
yeah, forgot to push. was commenting while making changes. It is pushed now.